middle_end/flambda2/simplify/simplify_unary_primitive.ml

```diff
@@ -657,12 +657,23 @@ let simplify_duplicate_array ~kind:_ ~(source_mutability : Mutability.t)
   (* This simplification should eliminate bounds checks on array literals. *)
   match source_mutability, destination_mutability with
   | Immutable, Mutable -> (
-    match T.meet_is_immutable_array (DA.typing_env dacc) arg_ty with
+    match T.meet_is_array (DA.typing_env dacc) arg_ty with
     | Invalid -> SPR.create_invalid dacc
-    | Need_meet ->
+    | Need_meet
+    | Known_result
+        { element_kind = _;
+          length = _;
+          contents = Known Mutable | Unknown;
+          alloc_mode = _
+        } ->
```

Comment:
We could put the `Known Mutable` case with `Invalid` (this is what happened before).
Although I don't think we actually expose `Pduparray`, so this case likely never occurs at all.